### PR TITLE
feat: Add replies to notification behavior (WEBAPP-5631)

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/notification/NotificationRepository.js [hunk 1]
```diff
@@ -832,7 +851,7 @@ z.notification.NotificationRepository = class NotificationRepository {
       return isEventToNotify;
     }
 
-    const isSelfMentioned = messageEntity.is_content() && messageEntity.isUserMentioned(userId);
-    return isEventToNotify && isSelfMentioned;
+    const isSelfMentionOrReply = messageEntity.is_content() && messageEntity.isResponseToUser(userId);
```
Comment (atomrc) [inline app/script/notification/NotificationRepository.js]: I guess the condition is a bit off here. Should probably be

```suggestion
    const isSelfMentionOrReply = messageEntity.is_content() && ( messageEntity.isUserMentioned(userId) || messageEntity.isResponseToUser(userId));
```

Comment (atomrc) [reply app/script/notification/NotificationRepository.js]: Ok, ignore this comment. It is included in the `isResponseToUser` actually

File: app/script/entity/message/ContentMessage.js [hunk 2]
```diff
@@ -117,7 +117,11 @@ z.entity.ContentMessage = class ContentMessage extends z.entity.Message {
   }
 
   isReplyToUser(userId) {
-    return this.quote() && this.quote().isQuoteFromUser(userId);
+    return this.quote() ? this.quote().isQuoteFromUser(userId) : false;
+  }
+
+  isResponseToUser(userId) {
```
Comment (atomrc) [inline app/script/entity/message/ContentMessage.js]: Not sure about the naming of this function. 
`isReply` and `isResponse` are very close and hard to distinguish. 

I believe we can come up with a better name. 

WDYT of `isTargetingUser`?

Comment (ffflorian) [reply app/script/entity/message/ContentMessage.js]: As agreed f2f, will change it to `isUserTargeted`.

**Pull request merged**